test/OpenTelemetry.Exporter.OneCollector.Tests/CommonSchemaJsonSerializationStateTests.cs

```diff
@@ -106,6 +139,48 @@ public void AddExtensionAttributeDuplicatesTest()
             json);
     }
 
+    [Fact]
+    public void AddExtensionAttributeMixedFieldAndPropertyTest()
+    {
+        using var stream = new MemoryStream();
+        using var writer = new Utf8JsonWriter(stream);
+
+        var state = new CommonSchemaJsonSerializationState("Test", writer);
+
+        state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext. foo .foo_field1", 1));
+        state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.foo", new List<KeyValuePair<string, object?>> { new KeyValuePair<string, object?>(" foo.field2 ", 2) }));
+        state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.foo .foo_field3 ", 3));
+
+        state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext. bar ", new Dictionary<string, object?>
+        {
+            ["   "] = 99, // Note: This will be dropped but the second entry will be added.
+            ["bar_field1"] = 1,
+        }));
+        state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext. bar .bar.field2", 2));
+        state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.bar.bar_field3", 3));
+
+        // Note: These fields should all be ignored.
+        state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.faz. ", 1));
+        state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.baz", new List<KeyValuePair<string, object?>> { new KeyValuePair<string, object?>(null!, 1) }));
+        state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.baz", new List<KeyValuePair<string, object?>> { new KeyValuePair<string, object?>(string.Empty, 2) }));
+        state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.baz", new List<KeyValuePair<string, object?>> { new KeyValuePair<string, object?>("    ", 3) }));
+        state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.jazz", "Only IReadOnlyList and IEnumerable maps are supported"));
+
+        Assert.Equal(4, state.ExtensionPropertyCount);
+        Assert.Equal(6, state.ExtensionAttributeCount);
+
+        writer.WriteStartObject();
+        state.SerializeExtensionPropertiesToJson(writeExtensionObjectEnvelope: true);
+        writer.WriteEndObject();
+        writer.Flush();
+
+        var json = Encoding.UTF8.GetString(stream.ToArray());
+
+        Assert.Equal(
+            "{\"ext\":{\"foo\":{\"foo_field1\":1,\"foo.field2\":2,\"foo_field3\":3},\"bar\":{\"bar_field1\":1,\"bar.field2\":2,\"bar_field3\":3}}}",
```

Comment:
Consider improving the readability - e.g. https://github.com/open-telemetry/opentelemetry-dotnet/blob/1da02973283654190c59d566df283c2072c58238/test/OpenTelemetry.Exporter.Prometheus.HttpListener.Tests/PrometheusSerializerTests.cs#L442

Reply:
Looks awesome, thanks!